Mulligan Round Fixes/Feature Requests

Incoming5643 opened the pull request:
Replaces the custom "how many people are dead?" check with the one used for survival rates on round end statistics because, lets be honest here, we know that one works.

Shuffles some flags around that should hopefully prevent round_converted() from running more than once. It's not a huge crisis if it does because late join antag is self capping, but still, not intended behavior!

Fixes #8089 (probably)

---

Adds an admin message on round conversion provided it gets far enough to matter. It also reports the same if no antags are made, to stop the exotic threat of unintentional admin meta.

fixes #8082

---

 Adds a lull between when the antagonist eats it and when the new antagonists are created: 3 ~ 7 minutes (5 average)

fixes #8105

Incoming5643 committed:
adds an admin message on round conversion provided it gets far enough to matter. It also reports the same if no antags are made, to stop the exotic threat of unintentional admin meta.

Iamgoofball commented:
might want to add a notification to round end stating it was a mulligan too, to also stop people bitching constantly about "fucking admins added traitors to a wizard/blob round!!!"

Ikarrus commented:
They'll get the point eventually @Iamgoofball

Incoming5643 committed:
Replaces the custom "how many people are dead?" check with the one used for survival rates on round end statitistics because, lets be honest here, we know that one works.

Shuffles some flags around that should hopefully prevent round_converted() from running more than once. It's not a huge crisis if it does because late join antag is self capping, but still, not intended behavoir!

Incoming5643 commented:
Added on some critical fixes for mulligan round antags. Please don't let this one rot in the pull queue too long

Menshin commented on code/game/gamemodes/game_mode.dm:
```diff
@@ -110,9 +110,9 @@
-			living_crew += mob
+	for(var/mob/Player in mob_list)
+		if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player))
+			living_crew++
```
No idea what _list = list + 1_ is doing in Byond.
Just to make sure it's what was intended to be written.

Incoming5643 replied:
Good catch

Incoming5643 committed:
Always remember to port smartly

Incoming5643 committed:
Adds a lull between when the antagonist eats it and when the new antagonists are created: 3 ~ 7 minutes (5 average)
